stackslib/src/config/mod.rs

```diff
@@ -1210,53 +1210,363 @@ impl std::default::Default for Config {
+    /// - "X2" for mainnet
+    /// - "T2" for testnet (xenon)
+    /// - Other values for specific test networks
+    ///
+    /// Configured as a 2-character ASCII string (e.g., "X2" for mainnet).
+    ///
+    /// Default: [`BLOCKSTACK_MAGIC_MAINNET`] (corresponds to "X2") unless overridden by [`BurnchainConfig::mode`] (e.g., "T2" for testnet (xenon)).
     pub magic_bytes: MagicBytes,
+    /// The public key associated with the local mining address for the underlying Bitcoin regtest node.
+    /// Provided as a hex string representing an uncompressed public key.
+    ///
+    /// It is primarily used in modes that rely on a controlled Bitcoin regtest backend
+    /// (e.g., "helium", "mocknet", "neon") where the Stacks node itself needs to
+    /// instruct the Bitcoin node to generate blocks.
+    ///
+    /// The key is used to derive the Bitcoin address that receives the coinbase rewards
+    /// when generating blocks on the regtest network.
+    ///
+    /// Mandatory if [`BurnchainConfig::mode`] is "helium".
+    /// This is intended strictly for testing purposes.
+    ///
+    /// Default: `None`
     pub local_mining_public_key: Option<String>,
+    /// Optional bitcoin block height at which the Stacks node process should gracefully exit.
+    /// When bitcoin reaches this height, the node logs a message and initiates a graceful shutdown.
+    ///
+    /// Applied only if [`BurnchainConfig::mode`] is not "mainnet".
+    /// This is intended strictly for testing purposes.
+    ///
+    /// Default: `None`
     pub process_exit_at_block_height: Option<u64>,
+    /// The interval, in seconds, at which the node polls the bitcoin node for new blocks and state updates.
+    ///
+    /// The default value of 10 seconds is mainly intended for testing purposes.
+    /// It's suggested to set this to a higher value for mainnet e.g. 300 seconds (5 minutes).
+    ///
+    /// Default: `10` seconds
     pub poll_time_secs: u64,
+    /// The default fee rate in satoshis per virtual byte (sats/vB) to use when estimating fees for miners
+    /// to submit bitcoin transactions (like block commits or leader key registrations).
+    ///
+    /// Only relevant if [`NodeConfig::miner`] is `true`.
+    ///
+    /// Default: [`DEFAULT_SATS_PER_VB`] (currently 50 satoshis per virtual byte)
     pub satoshis_per_byte: u64,
-    /// Maximum percentage of `satoshis_per_byte` that a Bitcoin fee rate may
-    /// be increased to when RBFing a transaction
+    /// Maximum fee rate multiplier allowed when using Replace-By-Fee (RBF) for bitcoin transactions.
+    /// Expressed as a percentage of the original [`BurnchainConfig::satoshis_per_byte`] rate (e.g.,
+    /// 150 means the fee rate can be increased up to 1.5x). Used in mining logic for RBF decisions
+    /// to cap the replacement fee rate.
+    ///
+    /// Only relevant if [`NodeConfig::miner`] is `true`.
+    ///
+    /// Default: [`DEFAULT_MAX_RBF_RATE`] (currently 150, i.e., 1.5x)
```

Comment:
I think if a default value is not a vec, we shouldn't use `[ value ]`. Same comment applies to other settings below using the same format. 
ex: https://github.com/Jiloc/stacks-core/blob/chore/document-node-config/stackslib/src/config/mod.rs#L1777-L1783
```suggestion
    /// Default: `DEFAULT_MAX_RBF_RATE` (currently 150, i.e., 1.5x)
```

Reply:
if that's the case, should the other PR's also use this format?